network/test/middleware_test.go

```diff
@@ -268,6 +268,45 @@ func (m *MiddlewareTestSuite) TestEcho() {
 	}
 }
 
+// TestSpoofedPubSubHello evaluates checking the originID of the message w.r.t. its libp2p network ID on PubSub
+// we check a pubsub message with a spoofed OriginID does not get delivered
+// This would be doubled with cryptographic verification of the libp2p network ID in production (see message signing options in pubSub initialization)
+func (m *MiddlewareTestSuite) TestSpoofedPubSubHello() {
+	first := 0
+	last := m.size - 1
+	firstNode := m.ids[first].NodeID
+	lastNode := m.ids[last].NodeID
+
+	// initially subscribe the nodes to the channel
+	for _, mw := range m.mws {
+		err := mw.Subscribe(testChannel)
+		require.NoError(m.Suite.T(), err)
+	}
+
+	// set up waiting for m.size pubsub tags indicating a mesh has formed
+	for i := 0; i < m.size; i++ {
+		select {
+		case <-m.obs:
+		case <-time.After(2 * time.Second):
+			assert.FailNow(m.T(), "could not receive pubsub tag indicating mesh formed")
+		}
+	}
+
+	var spoofedID flow.Identifier
+	copy(spoofedID[:16], firstNode[:16])
+	copy(spoofedID[16:], lastNode[16:])
```

Comment:
Are you just trying to create an invalid identifier here, or is there some reason you specifically take the first and last half of the identifiers of the first and last nodes?

If just trying to create an invalid identifier, you can just use `unittest.IdentifierFixture`

Also, I think we should test two separate cases here:
* the spoofed ID is actually a valid ID, but not the one that belongs to the sender
* the spoofed ID is an invalid ID that doesn't belong to anyone